src/hipscat/pixel_tree/pixel_alignment.py

```diff
@@ -82,138 +85,233 @@ def align_trees(
+    return PixelAlignment(PixelTree(result_tree, max_n), result_mapping, alignment_type)
 
 
-def _should_include_all_pixels_from_tree(tree_type: str, alignment_type: PixelAlignmentType) -> bool:
-    """If for a given alignment type, the left or right tree should include all pixels or just the
-    ones that overlap with the other tree.
+def get_pixel_mapping_df(mapping: np.ndarray, map_order: int) -> pd.DataFrame:
+    """Construct a DataFrame with HEALPix orders and pixels mapping left right and aligned pixels
 
     Args:
-        tree_type (str): 'left' for the left tree and 'right' for the right tree
-        alignment_type (PixelAlignmentType): The type of alignment being performed
+        mapping (np.ndarray): array of shape (6, len(aligned_pixels)) where the first two rows are the
+            intervals for the left pixels, the next two for right pixels, and the last two for aligned pixels
+        map_order (int): The HEALPix order of the intervals in the mapping array
 
     Returns:
-        A boolean indicating if the given tree type should include all pixels
+        A DataFrame with the orders and pixels of the aligned left and right pixels,
     """
-    left_add_types = [PixelAlignmentType.OUTER, PixelAlignmentType.LEFT]
-    right_add_types = [PixelAlignmentType.OUTER, PixelAlignmentType.RIGHT]
-    return (tree_type == LEFT_TREE_KEY and alignment_type in left_add_types) or (
-        tree_type == RIGHT_TREE_KEY and alignment_type in right_add_types
+    if len(mapping) > 0:
+        l_orders, l_pixels = get_pixels_from_intervals(mapping[0:2].T, map_order).T
+        r_orders, r_pixels = get_pixels_from_intervals(mapping[2:4].T, map_order).T
+        a_orders, a_pixels = get_pixels_from_intervals(mapping[4:6].T, map_order).T
+    else:
+        l_orders, l_pixels, r_orders, r_pixels, a_orders, a_pixels = [], [], [], [], [], []
+    result_mapping = pd.DataFrame.from_dict(
+        {
+            PixelAlignment.PRIMARY_ORDER_COLUMN_NAME: l_orders,
+            PixelAlignment.PRIMARY_PIXEL_COLUMN_NAME: l_pixels,
+            PixelAlignment.JOIN_ORDER_COLUMN_NAME: r_orders,
+            PixelAlignment.JOIN_PIXEL_COLUMN_NAME: r_pixels,
+            PixelAlignment.ALIGNED_ORDER_COLUMN_NAME: a_orders,
+            PixelAlignment.ALIGNED_PIXEL_COLUMN_NAME: a_pixels,
+        }
     )
+    result_mapping.replace(-1, None, inplace=True)
+    return result_mapping
 
 
-def _generate_pixel_mapping_from_tree(left: PixelTree, right: PixelTree, aligned: PixelTree) -> pd.DataFrame:
-    """Generates a pixel mapping dataframe from two trees and their aligned tree
+@njit(
+    numba.types.void(
+        numba.int64,
+        numba.int64,
+        numba.int64[:],
+        numba.boolean,
+        numba.types.List(numba.int64[::1]),
+    )
+)
+def _add_pixels_until(
+    add_from: int,
+    add_to: int,
+    matching_pix: np.ndarray,
+    is_left_pixel: bool,
+    mapping: List[np.ndarray],
+):
```

Comment:
The code coverage results are saying that most of this function is uncovered. It is probably worth adding a few tests for this specifically.

Reply:
You already addressed this with your comment about compiled code note being included in the code coverage computation.